proto_compat_test.go

```diff
@@ -180,18 +182,22 @@ func writeProtoBin(t *testing.T, m proto.Message, filename string) {
 	}
 }
 
-func writeProtoText(t *testing.T, m proto.Message, filename string) {
+func writeProtoText(t *testing.T, m protoreflect.ProtoMessage, filename string) {
 	t.Helper()
 
-	s := proto.MarshalTextString(m)
-	s = strings.Replace(strings.Replace(s, "<", "{", -1), ">", "}", -1)
+	b, err := prototext.Marshal(m)
+	if err != nil {
+		t.Fatal("problems marshaling proto message:", err)
+	}
 
-	if err := os.WriteFile(filename, []byte(s), 0644); err != nil {
+	b = bytes.Replace(bytes.Replace(b, []byte("<"), []byte("{"), -1), []byte(">"), []byte("}"), -1)
```

Comment:
I don't think this is needed anymore. I am not familiar with the old `proto.MarshalTextString`, but the new `prototext.Marshal` uses curly braces as delimiters: https://github.com/protocolbuffers/protobuf-go/blob/68463f0e96c93bc19ef36ccd3adfe690bfdb568c/encoding/prototext/encode.go#L117

Reply:
I see. I didn't realize that.
Now, this path is used to generate the compat test (proto) data using the old `https://github.com/golang/protobuf` implementation.